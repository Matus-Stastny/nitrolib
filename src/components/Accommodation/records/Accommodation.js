// @flow strict

import type { RoomType } from "../../../records/Room";

export type AccommodationType = {|
  hotelName: string,
  rating: number,
  location: {|
    latitude: number,
    longitude: number,
  |},
  address: string,
  photoUrl: string,
  rooms: Array<RoomType>,
  mapboxToken: string,
  zoom?: number,
  isMMB: boolean,
|};
