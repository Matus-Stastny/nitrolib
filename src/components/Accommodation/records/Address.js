// @flow strict

export type AddressType = ?{|
  +fullAddress: ?string,
|};
