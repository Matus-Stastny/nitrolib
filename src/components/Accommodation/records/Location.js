// @flow strict

export type LocationType = {|
  longitude: number,
  latitude: number,
  address: string,
  hotelName: string,
  mapboxToken: string,
  desktopWidth?: number,
  zoom?: number,
|};
