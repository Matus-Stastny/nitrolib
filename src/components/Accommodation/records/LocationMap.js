// @flow strict

export type LocationType = {|
  center: {|
    latitude: number,
    longitude: number,
  |},
  label: string,
  zoom: number,
  desktopWidth: number,
  mapboxToken: string,
|};
