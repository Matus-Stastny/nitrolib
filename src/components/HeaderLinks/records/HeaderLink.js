// @flow strict
export type HeaderLink = {|
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
|};
