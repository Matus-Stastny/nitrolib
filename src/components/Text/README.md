# Text

Our `Translate` wrapped in _Orbit_'s `Text`. Accepts both our and their props.

Useful for both translating and making text nicer!

Context needs:
* **intl**
