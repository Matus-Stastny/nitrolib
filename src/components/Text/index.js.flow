// @flow strict
import * as React from "react";

import type { Values } from "../../services/intl/translate";

type Props = {|
  t: string,
  values: Values,
  html: boolean,
|};

declare export default React.ComponentType<Props>;
