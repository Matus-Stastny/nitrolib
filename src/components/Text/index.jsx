// @flow strict
/* eslint-disable react/no-danger */
import * as React from "react";
import TextOrbit from "@kiwicom/orbit-components/lib/Text";
import type { Props as PropsOrbit } from "@kiwicom/orbit-components/lib/Text";

import Translate from "../Translate";
import type { Values } from "../../services/intl/translate";

type Props = {|
  ...$Diff<PropsOrbit, { children: React.Node }>,
  t: string,
  values?: Values,
  html?: boolean,
|};

const Text = ({ t, values, html, ...orbit }: Props) => (
  <TextOrbit {...orbit}>
    <Translate t={t} values={values} html={html} />
  </TextOrbit>
);

export default Text;
