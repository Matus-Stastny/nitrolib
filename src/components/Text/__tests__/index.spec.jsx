// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import { intlDefault } from "../../../records/Intl";

import Text from "..";

describe("#Text", () => {
  test("string", () => {
    const wrapper = shallow(<Text t="lol" />);

    expect(wrapper.prop("children")(intlDefault)).toMatchSnapshot();
  });

  test("html", () => {
    const wrapper = shallow(<Text t="lol" html />);

    expect(wrapper.prop("children")(intlDefault)).toMatchSnapshot();
  });
});
