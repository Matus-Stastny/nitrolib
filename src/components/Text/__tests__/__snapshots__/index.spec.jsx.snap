// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Text html 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "lol",
    }
  }
/>
`;

exports[`#Text string 1`] = `"lol"`;
